src/app/donation-start/donation-start-form/donation-tipping-slider/donation-tipping-slider.component.ts

```diff
@@ -0,0 +1,214 @@
+})
+export class DonationTippingSliderComponent implements OnInit, AfterContentInit, OnChanges, OnDestroy {
+
+
+  @Input() spaceBelow: 0 | 1 | 2 | 3 | 4 | 5 | 6 = 0;
+
+  @Input() colourScheme:
+  | 'primary' // blue
+  | 'secondary' // green
+  | 'tertiary' // coral
+  | 'brand-1' // cc-red
+  | 'brand-2' // wgmf-purple
+  | 'brand-3' // gmf-green
+  | 'brand-4' // emf-yellow
+  | 'brand-5' // c4c-orange
+  | 'brand-6' // mhf-turquoise
+  | 'white'
+  | 'black'
+  | 'grey-extra-light'
+  | 'grey-light'
+  | 'grey-medium'
+  | 'grey-dark' = 'primary';
+
+  @Input() percentageCurrent: number;
+  @Input() percentageStart: number;
+  @Input() percentageEnd: number;
+  @Input() donationAmount: number;
+  //ISO-4217 currency code (e.g. GBP, USD)
+  @Input() donationCurrency!: 'GBP' | 'USD';
+  @Input() onHandleMoved: (tipPercentage: number, tipAmount: number) => void;
+
+  @ViewChild('handle', {static: true}) handle: ElementRef;
+  @ViewChild('bar', {static: true}) bar: ElementRef;
+  @ViewChild('percentageValue', {static: true}) percentageWrap: ElementRef;
+  @ViewChild('donationValue', {static: true}) donationWrap: ElementRef;
+
+
+  containerClass: string;
+  derivedPercentage: number;
+  tipAmount: number;
+  currencyFormatted: string;
+
+  documentMouseupUnlistener: () => void;
+  documentMousemoveUnlistener: () => void;
+  documentTouchmoveUnlistener: () => void;
+
+  isMoving = false;
+  max: number;
+  pageX: any;
+  mousePos: number;
+  position: number;
+  disableDefaults: boolean = false;
+
+  constructor(
+    public renderer: Renderer2
+  ) {
+    this.documentMouseupUnlistener = renderer.listen('document', 'mouseup', () => {
+      this.isMoving = false;
+    });
+
+    this.documentMousemoveUnlistener = renderer.listen('document', 'mousemove', (event: MouseEvent | TouchEvent) => {
+      this.move(event);
+    });
+
+    this.documentTouchmoveUnlistener = renderer.listen('document', 'touchmove', (event: MouseEvent | TouchEvent) => {
+      this.move(event);
+    });
+  };
+
+
+  ngOnInit() {
+    this.containerClass = 'container space-below-' + this.spaceBelow;
+    this.calcAndSetPercentage();
+    this.adjustDonationPercentageAndValue();
+    this.updateHandlePosition(undefined)
+  }
+
+  ngAfterContentInit() {
+    if (this.bar) {
+      this.bar.nativeElement.addEventListener('mousedown',  (e: MouseEvent | TouchEvent) => {
+        this.isMoving = true;
+        this.move(e);
+      });
+
+      this.bar.nativeElement.addEventListener('touchstart', (e: MouseEvent | TouchEvent) => {
+        this.isMoving = true;
+        this.move(e);
+      });
+
+      this.bar.nativeElement.addEventListener('touchend', () => {
+        this.isMoving = false;
+      });
+    }
+  };
+
+  // detect changes in the donationAmount input
+  ngOnChanges(changed: any) {
+    this.max = this.bar.nativeElement.offsetWidth - this.handle.nativeElement.offsetWidth;
+    if (changed.donationAmount.currentValue != changed.donationAmount.previousValue) {
```

Comment:
I've still got my `any` allergy - and the comment seems redundant.

```suggestion
  ngOnChanges(changed: SimpleChanges) {
    this.max = this.bar.nativeElement.offsetWidth - this.handle.nativeElement.offsetWidth;
    if (changed.donationAmount!.currentValue != changed.donationAmount!.previousValue) {
```